debian/clearwater-socket-factory.postinst

```diff
@@ -0,0 +1,162 @@
+#
+# The author can be reached by email at clearwater@metaswitch.com or by
+# post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+#
+# Special Exception
+# Metaswitch Networks Ltd  grants you permission to copy, modify,
+# propagate, and distribute a work formed by combining OpenSSL with The
+# Software, or a work derivative of such a combination, even if such
+# copying, modification, propagation, or distribution would otherwise
+# violate the terms of the GPL. You must comply with the GPL in all
+# respects for all of the code used other than OpenSSL.
+# "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+# Project and licensed under the OpenSSL Licenses, or a work based on such
+# software and licensed under the OpenSSL Licenses.
+# "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+# under which the OpenSSL Project distributes the OpenSSL toolkit software,
+# as those licenses appear in the file LICENSE-OPENSSL.
+
+#
+# see: dh_installdeb(1)
+
+set -e
+
+# summary of how this script can be called:
+#        * <postinst> `configure' <most-recently-configured-version>
+#        * <old-postinst> `abort-upgrade' <new version>
+#        * <conflictor's-postinst> `abort-remove' `in-favour' <package>
+#          <new-version>
+#        * <postinst> `abort-remove'
+#        * <deconfigured's-postinst> `abort-deconfigure' `in-favour'
+#          <failed-install-package> <version> `removing'
+#          <conflicting-package> <version>
+# for details, see http://www.debian.org/doc/debian-policy/ or
+# the debian-policy package
+
+case "$1" in
+    configure)
+        # Stop the upstart jobs, if they are already running.  Note that stop
+        # will fail the script if they aren't runninh - hence the careful check
+        # below
+        if ( status clearwater-socket-factory-mgmt 2>&1 | grep start ); then
+          stop clearwater-socket-factory-mgmt
+        fi
+        if ( status clearwater-socket-factory-sig 2>&1 | grep start ); then
+          stop clearwater-socket-factory-sig
+        fi
+
+        # Stop any instances of the old socket factory (that attempted to do
+        # both namespaces) and delete the conf file if present.
+        if ( status clearwater-socket-factory 2>&1 | grep start ); then
+          stop clearwater-socket-factory
+        fi
+        rm -f /etc/init/clearwater-socket-factory.conf
+
+        # Create the config files for the management and signaling socket
+        # factories
+        cat <<EOF > /etc/init/clearwater-socket-factory-mgmt.conf
+# clearwater-socket-factory-mgmt
+#
+# This process runs as root in the management namespace, and enables processes
+# running in other namespaces to obtain sockets in this namespace.
+
+description "Clearwater management socket factory server"
+
+start on runlevel [2345]
+stop on runlevel [!2345]
+
+respawn
+
+script
+  WHITELIST_DIR=/etc/clearwater-socket-factory/management.d
+  NAMESPACE=management
+EOF
```

Comment:
Why do we do this rather than putting the files in place in the usual way?  Ditto below.